### PR TITLE
Upgrade Dotty version  to 0.22.0-RC1

### PR DESCRIPTION
**Pull request by gabro**
The upgrade went smooth, and I was able to remove the workaround for https://github.com/lampepfl/dotty/issues/7990, which is now fixed. 

By the way, I suppose we don't want to cross publish for different dotty versions, so I've just dropped 0.21.0-RC1.

### CODE REVIEW

**Comment by gabro** [comment]
Tests are failing on Scala Native with this error:

```
[error] /home/runner/work/munit/munit/munit/shared/src/main/scala/munit/Assertions.scala:10:8: object AssumptionViolatedException is not a member of package org.junit
[error] import org.junit.AssumptionViolatedException
[error]        ^
[error] /home/runner/work/munit/munit/munit/shared/src/main/scala/munit/Assertions.scala:41:19: not found: type AssumptionViolatedException
[error]         throw new AssumptionViolatedException(munitPrint(clue))
[error] 
```

@olafurpg any idea why?

**Comment by gabro** [comment]
Ah, looking at the build it's obvious: junit is not a dependency on native. Was that intentional? I know nothing about scala native, and it shows...

**Comment by gabro** [comment]
Ok, I turned my brain up and pushed a fix. It hopefully makes sense

File: munit/shared/src/main/scala/munit/Assertions.scala [hunk 1]
```diff
@@ -37,7 +38,7 @@ trait Assertions {
   )(implicit loc: Location): Unit = {
     StackTraces.dropInside {
       if (!cond) {
-        throw new DottyBugAssumptionViolatedException(munitPrint(clue))
```
Comment (olafurpg) [inline munit/shared/src/main/scala/munit/Assertions.scala]: Should we delete `DottyBugAssumptionViolatedException`?

Comment (olafurpg) [reply munit/shared/src/main/scala/munit/Assertions.scala]: Nevermind, the diff was collapsed. I see it's deleted now

Comment (gabro) [reply munit/shared/src/main/scala/munit/Assertions.scala]: didn't I do it?

**Pull request merged**